### DISCUSSION TITLE
Extending PayloadCMS Database Schema with Custom Types (e.g., tsvector)

### DISCUSSION THREAD

**Discussion by D0nte53** [payloadcms/payload#7472] [Q&A] [answered]
Hi,

I'm trying to implement full-text search in PayloadCMS as mentioned by [pooledg](https://payloadcms.com/community-help/github/best-way-to-implement-fulltext-search-on-a-collection).

 It seems we need to add columns, tables, views, or indexes after the `await payload.init({...})` statement in the `start()` function. However, when we use `payload.db.drizzle.execute(sql'...')`, PayloadCMS is no longer aware of the changes in the schema.

I'm looking for a way to avoid ``Warnings detected during schema push`` for fields and tables that I create with raw SQL using Drizzle.
Could you help me with a solution?

**Comment by r1tsuu** [comment, marked as answer]
The possible API for that is described here https://github.com/payloadcms/payload/discussions/4297#discussioncomment-10032148 as 3rd option from @DanRibbens 
I added mine suggestion too. I think it's really important to be able to customize the generated schema, not always you're fine with the generated variant.

Still, you can just completely ignore the schema push workflow via `db.push: false` and just use migrations workflow in the development as well. You create a migration -> add a change to the created migration file (for example tsvector type, or adding a custom table via `CREATE TABLE`) -> migrate.

**Reply by D0nte53** [reply]
Thank you very much for your response, I will do that. Yes, you are absolutely right; agreeing on a canonical way to do it could add great value to PayloadCMS !
